src/test/java/org/ibp/api/java/impl/middleware/inventory/manager/LotServiceImplTest.java

```diff
@@ -206,4 +240,75 @@ public void shouldSplitLot() {
+		lotGeneratorBatchRequestDto.getSearchComposite().setSearchRequest(searchOriginCompositeDto);
+
+		try {
+			this.lotService.createLots(LotServiceImplTest.PROGRAM_UUID, lotGeneratorBatchRequestDto);
+		} catch (final ApiRequestValidationException e) {
+			assertThat(Arrays.asList(e.getErrors().get(0).getCodes()), hasItem("search.origin.no.defined"));
+		}
+	}
+
+	@Test
+	public void testCreateLots_ThrowsException_WhenForGermplasmsearchTheSearchRequestIdNoReturnGids() {
+		final LotGeneratorBatchRequestDto lotGeneratorBatchRequestDto = buildLotGeneratorBatchRequestDto();
+		final SearchOriginCompositeDto searchOriginCompositeDto = new SearchOriginCompositeDto();
+		searchOriginCompositeDto.setSearchOrigin(SearchOriginCompositeDto.SearchOrigin.GERMPLASM_SEARCH);
+		lotGeneratorBatchRequestDto.getSearchComposite().setSearchRequest(searchOriginCompositeDto);
+
+		try {
+			this.lotService.createLots(LotServiceImplTest.PROGRAM_UUID, lotGeneratorBatchRequestDto);
+		} catch (final ApiRequestValidationException e) {
+			assertThat(Arrays.asList(e.getErrors().get(0).getCodes()), hasItem("searchrequestid.no.results"));
+		}
+	}
+
+	@Test
+	public void testCreateLots_ThrowsException_WhenForManageStudyTheSearchRequestIdNoReturnGids() {
+		final LotGeneratorBatchRequestDto lotGeneratorBatchRequestDto = buildLotGeneratorBatchRequestDto();
+		final SearchOriginCompositeDto searchOriginCompositeDto = new SearchOriginCompositeDto();
+		searchOriginCompositeDto.setSearchOrigin(SearchOriginCompositeDto.SearchOrigin.MANAGE_STUDY);
+		lotGeneratorBatchRequestDto.getSearchComposite().setSearchRequest(searchOriginCompositeDto);
+
+		try {
+			this.lotService.createLots(LotServiceImplTest.PROGRAM_UUID, lotGeneratorBatchRequestDto);
+		} catch (final ApiRequestValidationException e) {
+			assertThat(Arrays.asList(e.getErrors().get(0).getCodes()), hasItem("searchrequestid.no.results"));
+		}
+	}
+
+	@Test
+	public void testCreateLots_ThrowsException_WhenSearchRequestIsNullAndItemsAreInvalid() {
+		final Set items = Sets.newSet(new Random().nextInt());
+		final LotGeneratorBatchRequestDto lotGeneratorBatchRequestDto = buildLotGeneratorBatchRequestDto();
+		lotGeneratorBatchRequestDto.getSearchComposite().setItemIds(items);
+
+		final BindingResult errors = new MapBindingResult(new HashMap<>(), LotGeneratorBatchRequestDto.class.getName());
+		errors.reject("gids.invalid", new String[] {}, "");
+		Mockito.doThrow(new ApiRequestValidationException(errors.getAllErrors())).when(this.germplasmValidator)
+			.validateGids(ArgumentMatchers.any(BindingResult.class), ArgumentMatchers.any());
+
+		try {
+			this.lotService.createLots(LotServiceImplTest.PROGRAM_UUID, lotGeneratorBatchRequestDto);
+		} catch (final ApiRequestValidationException e) {
+			assertThat(Arrays.asList(e.getErrors().get(0).getCodes()), hasItem("gids.invalid"));
+		}
+	}
+
+	public static LotGeneratorBatchRequestDto buildLotGeneratorBatchRequestDto() {
```

Comment:
it does not need to be static. In fact, should be private and non static @cuenyad